protocol/messenger.go

```diff
@@ -659,6 +661,18 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if m.config.torrentConfig.Enabled {
+			adminCommunities, err := m.communitiesManager.Created()
+			if err == nil && len(adminCommunities) > 0 {
+				mailserverAvailable, unsubscribe := m.SubscribeMailserverAvailable()
+				go func() {
+					<-mailserverAvailable
+					close(unsubscribe)
+					m.InitHistoryArchiveTasks(adminCommunities)
```

Comment:
Here you can see how I wait for `mailserverAvailable` being emitted before we kick off the history archive tasks.

Again, there's probably a nicer more "go-like" way to do this.